src/network/test-networkd-conf.c

```diff
@@ -263,4 +265,214 @@ TEST(config_parse_match_strv) {
                                        "KEY3=val with \\quotation\\")));
 }
 
+static int parse_mpr(const char *rvalue, OrderedSet **nexthops) {
+        return config_parse_multipath_route(
+                        "network", "filename", 1, "section", 1, "MultiPathRoute", 0, rvalue, nexthops, NULL);
+}
+
+static void test_config_parse_multipath_route_one(const char *rvalue, int expected_ret, size_t expected_size) {
+        _cleanup_ordered_set_free_ OrderedSet *nexthops = NULL;
+        int r;
+
+        r = parse_mpr(rvalue, &nexthops);
+        ASSERT_EQ(r, expected_ret);
+        if (expected_ret > 0)
+                ASSERT_EQ(ordered_set_size(nexthops), expected_size);
+}
+
+static void test_config_parse_multipath_route_verify(
+                const char *rvalue,
+                int expected_family,
+                const char *expected_gw,
+                const char *expected_ifname,
+                int expected_ifindex,
+                uint32_t expected_weight) {
+        _cleanup_ordered_set_free_ OrderedSet *nexthops = NULL;
+        RouteNextHop *nh;
+
+        ASSERT_EQ(parse_mpr(rvalue, &nexthops), 1);
+        ASSERT_EQ(ordered_set_size(nexthops), 1u);
+
+        nh = ordered_set_first(nexthops);
+        ASSERT_NOT_NULL(nh);
+        ASSERT_EQ(nh->family, expected_family);
+
+        if (expected_gw) {
+                union in_addr_union gw;
+                ASSERT_OK(in_addr_from_string(expected_family, expected_gw, &gw));
+                ASSERT_EQ(memcmp(&nh->gw, &gw, FAMILY_ADDRESS_SIZE(expected_family)), 0);
+        } else {
+                ASSERT_FALSE(in_addr_is_set(nh->family, &nh->gw));
+        }
+
+        if (expected_ifname)
+                ASSERT_STREQ(nh->ifname, expected_ifname);
+        else
+                ASSERT_NULL(nh->ifname);
+
+        ASSERT_EQ(nh->ifindex, expected_ifindex);
+        ASSERT_EQ(nh->weight, expected_weight);
+}
+
+TEST(config_parse_multipath_route_basic) {
```

Comment:
I do not think it is necessary to split test cases. Please merge all `TEST(config_parse_multipath_route_xyz)`.